src/app/shared/models/protocol.ts

```diff
@@ -13,7 +13,20 @@ export interface Protocol {
   clinicalProtocol?: ClinicalProtocol
   onDemandProtocol?
   completionWindow?: TimeInterval
-  referenceTimestamp?: string
+  referenceTimestamp?: string | ProtocolReferenceTimestamp
+}
+
+export interface ProtocolReferenceTimestamp {
+  timestamp: string
+  format: ReferenceTimestampKey
+}
+
+export enum ReferenceTimestampKey {
```

Comment:
maybe rename to `ReferenceTimestampFormatKey`

Reply:
Or just `ReferenceTimestampFormat`?